### DISCUSSION TITLE
Looking for feedbacks to improve EPD documentation

### DISCUSSION THREAD

**Discussion by cardboardcode** [ros-industrial/easy_perception_deployment#40] [Q&A]
Hi all, 

This discussion thread has been created to capture any constructive feedbacks on how you would like to see the documentation be better at.

### The **current documentation** is displayed [here](https://epd-docs.readthedocs.io/en/latest/).

Based on the feedback you leave in this thread, the maintainer (mainly me @cardboardcode) may convert them to feature tickets where its implementation progress is tracked until it is completed.

### All **documentation codebase**  is accessible for public view [here](https://github.com/cardboardcode/epd_docs/tree/main).

Thank you for your time. :blush:

**Comment by Interpause** [comment]
Not related, but it is quite a coincidence the acronym EPD matches that of SUTD's Engineering & Product Design.

This is coming from the README mentioning the project is supported by the Singapore National Robotics Programme.

Any chance it is not a coincidence?

**Reply by cardboardcode** [reply]
Hi @Interpause, it is coincidence. 

I came up with the name **easy_perception_deployment** as a way to describe the ROS2 package function succintly while adhering to REP-144 Naming Conventions [[Link](https://www.ros.org/reps/rep-0144.html)].

I was not aware that SUTD has a department that shared the same name, up until now.

**Reply by Interpause** [reply]
Interesting. btw, I like your idea of being able to load onnx models via a GUI then deploy them as nodes easily. But to take it a few steps further, what if you used [Nividia's ISAAC DNN](https://github.com/NVIDIA-ISAAC-ROS/isaac_ros_dnn_inference) node for model inference, and removed the restrictions on the model input/output shape by coming up with some sort of user-configurable metadata specification/data structure adapter?

And about the SUTD thing, I am planning to enroll into SUTD's EPD once I ord, so I couldn't help but comment when I came across this repository while researching alternatives to my own code :p.

**Reply by cardboardcode** [reply]
Noted. Thank you for the suggestion. Will look into it when I have the bandwidth. 

Currently, EPD still targets **x64**-based systems, rather **arm**-based systems, although there may be potential plans to start migrating towards **arm** devices. Will update here probably when there are further confirmations made within the dev team.

Do let me know if there are anything I can do to avoid our codebases overlapping too much in functionality in order to preserve our respective uniqueness.
